### PR TITLE
Avoid Inbox Note cleanup during admin AJAX

### LINKED ISSUE #68019: Avoid WooCommerce Admin Inbox Note cleanup during admin AJAX
## Problem

WooCommerce 11.0.1 registers two Inbox Note cleanup paths on `admin_init`. WordPress runs `admin_init` while serving `wp-admin/admin-ajax.php`, so unrelated authenticated AJAX requests reach these operations without an AJAX guard:

1. `src/Blocks/Domain/Bootstrap.php:94-103` → `src/Blocks/InboxNotifications.php:20-21` → `Notes::delete_notes_with_name( 'surface_cart_checkout' )`.
2. `src/Admin/Notes/Notes.php:25-29,300-310` → `DataStore::get_note_ids_by_type( 'survey' )` in `src/Admin/Notes/DataStore.php:548-555` → hydrate survey notes and update actioned rows as deleted.

A real authenticated AJAX request against the stock 11.0.1 package produced two `wc_admin_notes` SELECTs on every request: one name lookup and one survey-type lookup. Matching rows can add hydration/action queries and deletes or updates.

Audited artifact: [WooCommerce 11.0.1](<https://downloads.wordpress.org/plugin/woocommerce.11.0.1.zip>), SHA-256 `da189b6616c610d15a2106f93151dab81b78f83e075bcefce221ac0d00b4fa21`.

This is longstanding performance debt, not a recent regression. The survey callback has been AJAX-reachable since [WooCommerce Admin #4686](<https://github.com/woocommerce/woocommerce-admin/pull/4686>) shipped in WooCommerce 4.4.0; its current broad query dates to [#8206](<https://github.com/woocommerce/woocommerce-admin/pull/8206>) and WooCommerce 6.5.0. The surface-note lookup first shipped in [WooCommerce Blocks #4518](<https://github.com/woocommerce/woocommerce-blocks/pull/4518>) with WooCommerce 5.8.0, and its current cleanup form dates to [#6805](<https://github.com/woocommerce/woocommerce-blocks/pull/6805>) and WooCommerce 6.9.0.

As supporting scale context, [WooCommerce.com](<http://WooCommerce.com>) New Relic telemetry for August 24, 2026 UTC recorded 112,604 `wp_wc_admin_notes` SELECT measurements across 112,780 admin-AJAX transactions (99.84%). They averaged 4.30 ms and totaled 483.9 database-seconds—about 4.97% of admin-AJAX database duration and 0.755% of total admin-AJAX transaction duration. This is aggregate table/operation/scope telemetry, not raw SQL or a backtrace, so it is not callback-attributable and the duration is an upper bound. It does not reproduce the stock two-query result one-for-one; the stock test above provides the direct call-path evidence.

## Acceptance Criteria

- [ ] Unrelated admin-AJAX requests no longer execute either Inbox Note cleanup path or their note-table queries/writes.
- [ ] Cleanup of the stale `surface_cart_checkout` note still occurs at an appropriate lifecycle point.
- [ ] Cleanup of actioned survey notes still occurs at an appropriate scheduled or state-transition point.
- [ ] Regression tests cover admin AJAX and the intended non-AJAX cleanup behavior.

## Technical Notes

A minimal stopgap is to return from both `admin_init` callbacks when `wp_doing_ajax()` is true. Prefer moving the surface-note cleanup to a versioned upgrade migration and the survey cleanup to `wc_admin_daily` or the relevant note-action transition.

### PR DESCRIPTION
**Pull request by NeosinneR**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

WordPress fires `admin_init` for admin AJAX requests, so unrelated requests currently run two Inbox Note cleanup paths. This moves retired Surface Cart and Checkout note removal to the WooCommerce 11.2 update and moves actioned survey-note cleanup from `admin_init` to `wc_admin_daily`.

Actioned survey notes may remain until the next daily task runs, or longer if scheduled actions are delayed. No public PHP signatures or hook arguments change, but code that relied on cleanup as an `admin_init` side effect will observe different timing. Both paths use the current site's note tables; multisite was not tested manually.

Closes #68019.

Closes WOOPLUG-7598

### How to test the changes in this Pull Request:

1. Create the retired Surface Cart and Checkout note, an actioned survey note, an unactioned survey note, and an actioned non-survey note.
2. Trigger `admin_init`, including through an unrelated authenticated admin AJAX request, and confirm none of those notes are deleted.
3. Run the WooCommerce 11.2 update callback and confirm only the retired Surface Cart and Checkout note is deleted. Run it again and confirm it is idempotent.
4. Trigger `wc_admin_daily` and confirm only the actioned survey note is deleted.

### Testing that has already taken place:

The targeted Notes, update, and Blocks bootstrap suites pass. PHPStan and both required PHP lint passes are clean. Codex and Claude independently reviewed the rebased branch; valid minor test-coverage findings were fixed and no blocking issues remain.

### Milestone

- [x] Automatically assign milestone for the **[next WooCommerce version](../blob/trunk/plugins/woocommerce/woocommerce.php#L6)**

> **Note:** Check the box above to have the milestone automatically assigned when merged.
> Alternatively (e.g. for point releases), manually assign the appropriate milestone.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.
-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Comment</summary>

#### Comment

A changelog entry is included in this branch.

</details>

### Release Communication

Select the release communication labels this PR needs:

- [ ] **Feature Highlight** - For user-facing features (what changed, user impact)
- [ ] **Developer Advisory** - For developer-facing changes (what changed, how to detect, actions needed)

### Use of AI Tools

Codex implemented and tested these changes and drafted this Pull Request. Claude Code performed an independent code review. The contributor reviewed the generated output and accepts responsibility for the changes.

### CODE REVIEW

**Comment by NeosinneR** [comment]
@cursor review

**Comment by NeosinneR** [comment]
@cursor review

File: plugins/woocommerce/tests/php/includes/wc-update-functions-test.php [hunk 1]
```diff
@@ -468,4 +471,31 @@ public function test_wc_update_1120_migrate_stock_notifications_alpha_constant_f
...
+		$this->assertContains( 'wc_update_1120_delete_surface_cart_checkout_note', $db_updates['11.2.0'] );
+
+		$note = new Note();
+		$note->set_name( InboxNotifications::SURFACE_CART_CHECKOUT_NOTE_NAME );
+		$note->set_title( 'Surface Cart and Checkout' );
+		$note->set_content( 'Test content' );
+		$note->set_type( Note::E_WC_ADMIN_NOTE_INFORMATIONAL );
+		$note->set_source( 'PHPUNIT_TEST' );
+		$note->add_action( 'learn-more', 'Learn more', 'https://woocommerce.com/' );
+		$note->save();
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply plugins/woocommerce/tests/php/includes/wc-update-functions-test.php:521]: @coderabbitai fixed in https://github.com/woocommerce/woocommerce/commit/413315aaf536cf7c4b3ba41ba67a4ae1c1ebe9b0

File: plugins/woocommerce/tests/php/src/Blocks/Domain/BootstrapTest.php [hunk 2]
```diff
@@ -88,6 +88,34 @@ private function set_register_blocks_has_run_flag( bool $has_run ): void {
...
+				}
+
+				$this->assertNotSame(
+					$bootstrap_file,
+					wp_normalize_path( (string) $reflection->getFileName() ),
+					'Bootstrap callbacks on admin_init also run during unrelated admin AJAX requests.'
+				);
+			}
+		}
+	}
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply plugins/woocommerce/tests/php/src/Blocks/Domain/BootstrapTest.php:115]: @copilot fixed in https://github.com/woocommerce/woocommerce/commit/413315aaf536cf7c4b3ba41ba67a4ae1c1ebe9b0

File: plugins/woocommerce/includes/wc-update-functions.php [hunk 3]
```diff
@@ -3703,3 +3704,14 @@ function wc_update_1120_migrate_stock_notifications_alpha_constant() {
...
+
+/**
+ * Delete the retired Surface Cart and Checkout inbox note.
+ *
+ * @since 11.2.0
+ *
+ * @return void
+ */
+function wc_update_1120_delete_surface_cart_checkout_note(): void {
+	InboxNotifications::delete_surface_cart_checkout_blocks_notification();
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply plugins/woocommerce/includes/wc-update-functions.php:3716]: @coderabbitai fixed in https://github.com/woocommerce/woocommerce/commit/413315aaf536cf7c4b3ba41ba67a4ae1c1ebe9b0

File: plugins/woocommerce/includes/class-wc-install.php [hunk 4]
```diff
@@ -353,6 +354,7 @@ class WC_Install {
 		'11.2.0'   => array(
 			'wc_update_1120_remove_abandoned_cart_recovery',
 			'wc_update_1120_migrate_stock_notifications_alpha_constant',
+			'WC_Install::delete_surface_cart_checkout_note',
```
[the remark this answers is not in the document]
Comment (NeosinneR) [reply plugins/woocommerce/includes/class-wc-install.php]: @coderabbitai false positive, because this callback only existed in this unmerged PR and has never shipped in a WooCommerce release, so no production site can already have it queued or depend on it as a public symbol.

File: plugins/woocommerce/src/Admin/Notes/Notes.php [hunk 5]
```diff
@@ -24,7 +24,7 @@ class Notes {
 	 */
 	public static function init() {
 		add_action( 'admin_init', array( __CLASS__, 'schedule_unsnooze_notes' ) );
-		add_action( 'admin_init', array( __CLASS__, 'possibly_delete_survey_notes' ) );
+		add_action( 'wc_admin_daily', array( __CLASS__, 'possibly_delete_survey_notes' ) );
```
Comment (xristos3490) [inline plugins/woocommerce/src/Admin/Notes/Notes.php]: Kinda nit: Since this is more of a cleanup, we might want to add it with a lower priority, like `9`, to clear the execution order of other related daily tasks, like the remote-polling task. So, cleanup first, fetch after.

**Review by xristos3490 — APPROVED** [review]
Nice work, @NeosinneR! 👌 

This looks good to me! Left a nit, but pre-approving.

File: plugins/woocommerce/src/Admin/Notes/Notes.php (hunk 5 above)
Comment (NeosinneR) [reply plugins/woocommerce/src/Admin/Notes/Notes.php]: Good thinking, https://github.com/woocommerce/woocommerce/pull/68088/changes/a912e8a53c0cfebd29dc4742370d785dca969517
